lms/djangoapps/badges/migrations/0002_assertions_transfer.py

```diff
@@ -0,0 +1,233 @@
+# -*- coding: utf-8 -*-
+import json
+
+import os
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        from django.core.files.base import ContentFile
+        from xmodule.modulestore.django import modulestore
+        from badges.events import course_complete
+        classes = {}
+        for badge in orm['certificates.BadgeAssertion'].objects.all():
+            if (badge.course_id, badge.mode) not in classes:
+                course = modulestore().get_course(badge.course_id)
+                image_config = orm['certificates.BadgeImageConfiguration'].objects.get(mode=badge.mode)
+                icon = image_config.icon
+                badge_class = orm.BadgeClass(
+                    display_name=course.display_name,
+                    criteria=course_complete.evidence_url(badge.user_id, badge.course_id),
+                    description=course_complete.badge_description(course, badge.mode),
+                    slug=course_complete.course_slug(badge.course_id, badge.mode),
+                    mode=image_config.mode,
+                    course_id=badge.course_id,
+                )
+                file_content = ContentFile(icon.read())
+                badge_class._meta.get_field('image').generate_filename = \
+                    lambda inst, fn: os.path.join('badge_classes', fn)
+                badge_class.image.save(icon.name, file_content)
+                badge_class.save()
+                classes[(badge.course_id, badge.mode)] = badge_class
+            orm.BadgeAssertion(
+                user_id=badge.user_id,
+                badge_class=classes[(badge.course_id, badge.mode)],
+                data=json.loads(badge.data),
+                backend='BadgrBackend',
+                image_url=badge.data['image'],
+                assertion_url=badge.data['json']['id'],
+            ).save()
+
+        for configuration in orm['certificates.BadgeImageConfiguration'].objects.all():
+            file_content = ContentFile(configuration.icon.read())
+            new_conf = orm.CourseCompleteImageConfiguration(
+                default=configuration.default,
+                mode=configuration.mode,
+            )
+            new_conf.icon.save(configuration.icon.name, file_content)
+            new_conf.save()
+
+    def backwards(self, orm):
+        "Write your backwards methods here."
```

Comment:
If we don't implement the backwards migration, shouldn't we be throwing an exception here?  Or even better, shouldn't we just implement the backwards migration?  I find it really helpful to be able to migrate my devstack back and forth between versions sometimes.